polyfills/localStorage/detect.js

```diff
@@ -1 +1,10 @@
-'localStorage' in this
+'localStorage' in this && (function() {
+	try {
+		// Needed for Safari private browsing.
+		this.localStorage.setItem('storage_test', 1);
+		this.localStorage.removeItem('storage_test');
+		return true;
+	} catch (e) {
+		return false;
+	}
+}())
```

Comment:
The last thing I'd suggest here is using `.call(this)`, that way you're accessing `localStorage` the same way throughout the entire feature check.

```js
'localStorage' in this && (function() {
	try {
		// Needed for Safari private browsing.
		this.localStorage.setItem('storage_test', 1);
		this.localStorage.removeItem('storage_test');
		return true;
	} catch (e) {
		return false;
	}
}).call(this);
```

Reply:
Good point! I've amended this is a subsequent commit